Do not force QT_WAYLAND_DISABLE_WINDOWDECORATION=1 on wayland greeters

aleixpol opened the pull request:
This would put Qt in a weird state (see the issue that we are fixing) where it would use the wrong GL configuration.

Arguably it's an error of Qt for having a bogus default pixel format but this change should fix it but also make sure it behaves like other Qt applications and thus more predictably.

Fixes #1441

aleixpol committed:
Do not force QT_WAYLAND_DISABLE_WINDOWDECORATION=1 on wayland greeters

This would put Qt in a weird state (see the issue that we are fixing)
where it would use the wrong GL configuration.

Arguably it's an error of Qt for having a bogus default pixel format but
this change should fix it but also make sure it behaves like other Qt
applications and thus more predictably.

Fixes #1441

Vogtinator commented:
I would assume that at the point it was added, setting the variable was necessary. Did that change or is it still the case with `fullscreen-shell-v1`?

(IMO `QT_WAYLAND_SHELL_INTEGRATION` should also no longer be set - with #1673, the default xdg-shell should also work)

Vogtinator committed:
Process all available auth messages in a loop

When QLocalSocket::readyRead is emitted, it's important to read all available
data. Otherwise the event loop is reentered while data is still available and
readyRead won't be emitted again.

This can happen in practice if sddm-helper sends a message which does not
need a reply (e.g. INFO) followed by another message before the daemon
processed the first one. In that case the daemon would never process the second
one, potentially leaving sddm-helper waiting for a reply.

aleixpol commented:
> I would assume that at the point it was added, setting the variable was necessary. Did that change or is it still the case with `fullscreen-shell-v1`?

I would assume as well. It makes no sense that Qt is rendering decorations on fullscreen-shell-v1 either, we can consider it a limitation of its impementation in Qt.

In practice, I don't think anyone should be using the default behaviour of weston + fullscreen-shell-v1 anyway.

Vogtinator reviewed (approved):
I just tried this with weston, works fine without it :shrug:

Vogtinator commented:
For some reason [48a98c7](https://github.com/sddm/sddm/pull/1699/commits/48a98c7ba45b49cf443ac46f0d218f044933c603) shows up as part of this PR now even though that's in the target branch. I'll hit the button anyway.
